atlasdb-commons/src/main/java/com/palantir/common/proxy/ExperimentRunningProxy.java

```diff
@@ -36,42 +37,48 @@
 public final class ExperimentRunningProxy<T> extends AbstractInvocationHandler {
     private static final Logger log = LoggerFactory.getLogger(ExperimentRunningProxy.class);
     static final Duration REFRESH_INTERVAL = Duration.ofMinutes(10);
+    static final Duration CLIENT_REFRESH_INTERVAL = Duration.ofMinutes(30);
 
-    private final T experimentalService;
+    private final AtomicReference<T> experimentalService;
+    private final Supplier<T> experimentalServiceSupplier;
```

Comment:
thoughts on just making this a `Suppliers.memoizeWithExpiration`? the semantics are the same no and should be less intrusive?

Reply:
Derp, I wanted to make it smarter, but then changed my mind and forgot there was an easy way to do it